src/Infrastructure/GoogleListingsAndAdsPlugin.php

```diff
@@ -88,6 +94,11 @@ public function register(): void {
 			self::SERVICE_REGISTRATION_HOOK,
 			function() {
 				$this->maybe_register_services();
+
+				// Check if activation source is set.
+				if ( ! $this->options->get( OptionsInterface::TRACKED_ACTIVATION_SOURCE ) ) {
+					$this->container->get( ActivatedEvents::class )->maybe_track_activation_source();
+				}
```

Comment:
Could you clarify why this is done here?

Based on the discussions, I thought our intention was to store an option `PLUGIN_ACTIVATED`, which would stay false until the activate function has run with the requirement check passing. So I'd be expecting to see something like this here:
```php
if ( ! $this->options->get( OptionsInterface::PLUGIN_ACTIVATED ) ) {
	$this->activate();
}
```

And then within the activate function we check something like `PluginValidator::validated()` and have it run the activation and update the option if the conditions are met. Something like that would be forcing the activation code to be run (albeit delayed) after dependencies have been met.

Reply:
Thanks, as per our previous discussion. I implemented the PLUGIN_ACTIVATED flag. I am checking the flag in the init hook to avoid issues with wpdb:$action_scheduler that only get checked on init.